src/serde_seq.rs

```diff
@@ -0,0 +1,112 @@
+//! Functions to serialize and deserialize an `IndexMap` as an ordered sequence.
+//!
+//! The default `serde` implementation serializes `IndexMap` as a normal map,
+//! but there is no guarantee that serialization formats will preserve the order
+//! of the key-value pairs. This module serializes `IndexMap` as a sequence of
+//! `(key, value)` elements instead, in order.
```

Comment:
I'm not sure whether serializing tuples is a good choice, although it's the easiest to implement. Alternatively, serde sequences may be heterogeneous, so we could use a flat sequence of `key0, value0, key1, value1, ...` instead. I don't know whether all serialization formats support that though.

Reply:
This is also just a helper, so we don't have to be perfect here. A third party can write a similar custom module with different choices, as there aren't issues like trait coherence for this.

Reply:
Yet another possibility is a pair of sequences, `([key, ...], [value, ...])`, but when deserializing we would have to read keys into temporary storage before we start reading values and insert into the map.